data/mysql/schema.sql

```diff
@@ -1601,7 +1601,7 @@ CREATE TABLE `ezprest_authcode` (
   `user_id` int(11) NOT NULL DEFAULT '0',
   PRIMARY KEY (`id`),
   KEY `authcode_client_id` (`client_id`)
-) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_520_ci;
```

Comment:
_VARCHAR(M): M represents the maximum column length in characters. The range of M is 0 to 65,535. The effective maximum length of a VARCHAR is subject to the maximum row size (65,535 bytes, which is shared among all columns) and the character set used. For example, utf8 characters can require up to three bytes per character, so a VARCHAR column that uses the utf8 character set can be declared to be a maximum of 21,844 characters._
https://mariadb.com/kb/en/library/varchar/

This is varchar(200). Since mariadb reads this as length in chars, then 200 x 4 = 800 bytes which overflows the 767 bytes limit. Where that limit comes from I don't know - is it a generic key length limit for strings?

Reply:
Maybe wrong. Mysql also says it reads it as length in characters, not bytes, so we should get the same error there, except we don't. 😕 
https://dev.mysql.com/doc/refman/5.7/en/char.html

Reply:
The 767 bytes limit is a hard limit on innodb indexes. So it seems we can't have indexes on varchars bigger than 191. (191 x 4 = 764).
https://stackoverflow.com/questions/1814532/1071-specified-key-was-too-long-max-key-length-is-767-bytes?answertab=votes#tab-top

But we can't just willy nilly truncate all indexed varchars to 191 either - many are 255 chars long. No idea why this doesn't break the tests, but anyway... dang. 😖